### PR TITLE
SUS-1801: Don't abuse MASTER_POS_WAIT when obtaining a slave connection

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-1801

### CODE REVIEW

**Comment by drozdo** [comment]
LGTM

**Review by macbre — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: includes/db/LoadBalancer.php [hunk 1]
```diff
@@ -341,13 +341,8 @@ function getReaderIndex( $group = false, $wiki = false ) {
...
 		if ( $i !== false ) {
+			# Wikia change - SUS-1801: don't abuse MASTER_POS_WAIT when obtaining slave connection
 			# Slave connection successful
-			# Wait for the session master pos for a short time
-			if ( $this->mWaitForPos && $i > 0 ) {
-				if ( !$this->doWait( $i ) ) {
-					$this->mServers[$i]['slave pos'] = $conn->getSlavePos();
-				}
-			}
 			if ( $this->mReadIndex <=0 && $this->mLoads[$i]>0 && $i !== false ) {
```
Comment (WikiaTech) [inline includes/db/LoadBalancer.php:346]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues
* ![MAJOR][MAJOR] 13 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 7 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [LoadBalancer.php](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php): File "LoadBalancer.php" has 1153 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L15](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L15): Class "LoadBalancer" has 42 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L182](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L182): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L198](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L198): The Cyclomatic Complexity of this function "getReaderIndex" is 27 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L198](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L198): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L198](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L198): This function "getReaderIndex" has 155 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L201](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L201): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L209](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L209): Replace "and" with "&&". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2010)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L379](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L379): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L424](https://github.com/Wikia/app/blob/8c1548dae1e7f7014c0d10d75947631d4c9df1dc/includes/db/LoadBalancer.php#L424): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues
* ![MAJOR][MAJOR] 13 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 7 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [LoadBalancer.php](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php): File "LoadBalancer.php" has 1153 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L15](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L15): Class "LoadBalancer" has 42 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L182](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L182): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L198](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L198): The Cyclomatic Complexity of this function "getReaderIndex" is 27 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L198](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L198): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L198](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L198): This function "getReaderIndex" has 155 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L201](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L201): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L209](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L209): Replace "and" with "&&". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2010)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L379](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L379): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [LoadBalancer.php#L424](https://github.com/Wikia/app/blob/1b243d882bf3c40206a387404b2e8106cbd51af1/includes/db/LoadBalancer.php#L424): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**